server/src/main/java/org/apache/iotdb/db/query/dataset/ShowTimeseriesDataSet.java

```diff
@@ -19,44 +19,123 @@
+      new PartialPath(COLUMN_TIMESERIES_DATATYPE, false), new PartialPath(COLUMN_TIMESERIES_ENCODING, false),
+      new PartialPath(COLUMN_TIMESERIES_COMPRESSION, false), new PartialPath(COLUMN_TAG, false),
+      new PartialPath(COLUMN_ATTRIBUTE, false)};
+  private static TSDataType[] resourceTypes = {TSDataType.TEXT, TSDataType.TEXT, TSDataType.TEXT,
+      TSDataType.TEXT, TSDataType.TEXT, TSDataType.TEXT, TSDataType.TEXT, TSDataType.TEXT};
+
+  public ShowTimeseriesDataSet(ShowTimeSeriesPlan showTimeSeriesPlan, QueryContext context,
+      List<ShowTimeSeriesResult> timeseriesList) {
+    super(Arrays.asList(resourcePaths), Arrays.asList(resourceTypes));
     this.plan = showTimeSeriesPlan;
     this.context = context;
+    this.timeseriesList = timeseriesList;
+    this.hasSetRecord = false;
+  }
+
+  public QueryDataSet getQueryDataSet() {
+    hasLimit = plan.hasLimit();
+    for (ShowTimeSeriesResult result : timeseriesList) {
+      RowRecord record = new RowRecord(0);
+      updateRecord(record, result.getName());
+      updateRecord(record, result.getAlias());
+      updateRecord(record, result.getSgName());
+      updateRecord(record, result.getDataType().toString());
+      updateRecord(record, result.getEncoding().toString());
+      updateRecord(record, result.getCompressor().toString());
+      updateRecord(record, result.getTag());
+      updateRecord(record, result.getAttribute());
+      putRecord(record);
+    }
+    return this;
+  }
+
+  public List<RowRecord> transferShowTimeSeriesResultToRecordList(
+      List<ShowTimeSeriesResult> timeseriesList) {
+    List<RowRecord> records = new ArrayList<>();
+    for (ShowTimeSeriesResult result : timeseriesList) {
+      RowRecord record = new RowRecord(0);
+      updateRecord(record, result.getName());
+      updateRecord(record, result.getAlias());
+      updateRecord(record, result.getSgName());
+      updateRecord(record, result.getDataType().toString());
+      updateRecord(record, result.getEncoding().toString());
+      updateRecord(record, result.getCompressor().toString());
+      updateRecord(record, result.getTag());
+      updateRecord(record, result.getAttribute());
+      records.add(record);
+    }
+    return records;
+  }
+
+  private void updateRecord(RowRecord record, Map<String, String> map) {
+    String text = map.entrySet().stream()
+        .map(e -> "\"" + e.getKey() + "\"" + ":" + "\"" + e.getValue() + "\"")
+        .collect(Collectors.joining(","));
+
+    updateRecord(record, text.length() == 0 ? null : "{" + text + "}");
+  }
+
+  private void updateRecord(RowRecord record, String s) {
+    if (s == null) {
+      record.addField(null);
+      return;
+    }
+    Field field = new Field(TSDataType.TEXT);
+    field.setBinaryV(new Binary(s));
+    record.addField(field);
   }
 
   @Override
   protected boolean hasNextWithoutConstraint() throws IOException {
+    if (!hasSetRecord) {
```

Comment:
i think that this code can be optimizing:
1. Take the data directly from the `MManager` and initialize it in the `constructor`
2. if the total amount of data is less than the `fetchSize`, which means there is no more data, reducing the access to the `MManager`.

> when hasLimit = false, limit was means fetchSize

3. if `index == result.size()== fetchSize`, get data from `MManager` again.